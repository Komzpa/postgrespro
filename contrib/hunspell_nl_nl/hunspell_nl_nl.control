# hunspell_nl_nl extension
comment = 'Dutch Hunspell Dictionary'
default_version = '1.0'
module_pathname = '$libdir/hunspell_nl_nl'
relocatable = true