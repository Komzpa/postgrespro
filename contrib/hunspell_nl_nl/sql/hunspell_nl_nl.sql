CREATE EXTENSION hunspell_nl_nl;

CREATE TABLE table1(name varchar);
INSERT INTO table1 VALUES ('deuren'), ('deurtje'), ('deur'),
						('twee'), ('tweehonderd'), ('tweeduizend');

SELECT d.* FROM table1 AS t, LATERAL ts_debug('public.dutch', t.name) AS d;

CREATE INDEX name_idx ON table1 USING GIN (to_tsvector('public.dutch', "name"));
SELECT * FROM table1 WHERE to_tsvector('public.dutch', name)
	@@ to_tsquery('public.dutch', 'deurtje');
SELECT * FROM table1 WHERE to_tsvector('public.dutch', name)
	@@ to_tsquery('public.dutch', 'twee');

DROP INDEX name_idx;
CREATE INDEX name_idx ON table1 USING GIST (to_tsvector('public.dutch', "name"));
SELECT * FROM table1 WHERE to_tsvector('public.dutch', name)
	@@ to_tsquery('public.dutch', 'deurtje');
SELECT * FROM table1 WHERE to_tsvector('public.dutch', name)
	@@ to_tsquery('public.dutch', 'twee');