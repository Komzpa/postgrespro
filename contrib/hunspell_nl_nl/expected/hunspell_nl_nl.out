CREATE EXTENSION hunspell_nl_nl;
CREATE TABLE table1(name varchar);
INSERT INTO table1 VALUES ('deuren'), ('deurtje'), ('deur'),
						('twee'), ('tweehonderd'), ('tweeduizend');
SELECT d.* FROM table1 AS t, LATERAL ts_debug('public.dutch', t.name) AS d;
   alias   |   description   |    token    |        dictionaries         |   dictionary   | lexemes 
-----------+-----------------+-------------+-----------------------------+----------------+---------
 asciiword | Word, all ASCII | deuren      | {dutch_hunspell,dutch_stem} | dutch_hunspell | {deur}
 asciiword | Word, all ASCII | deurtje     | {dutch_hunspell,dutch_stem} | dutch_hunspell | {deur}
 asciiword | Word, all ASCII | deur        | {dutch_hunspell,dutch_stem} | dutch_hunspell | {deur}
 asciiword | Word, all ASCII | twee        | {dutch_hunspell,dutch_stem} | dutch_hunspell | {twee}
 asciiword | Word, all ASCII | tweehonderd | {dutch_hunspell,dutch_stem} | dutch_hunspell | {twee}
 asciiword | Word, all ASCII | tweeduizend | {dutch_hunspell,dutch_stem} | dutch_hunspell | {twee}
(6 rows)

CREATE INDEX name_idx ON table1 USING GIN (to_tsvector('public.dutch', "name"));
SELECT * FROM table1 WHERE to_tsvector('public.dutch', name)
	@@ to_tsquery('public.dutch', 'deurtje');
  name   
---------
 deuren
 deurtje
 deur
(3 rows)

SELECT * FROM table1 WHERE to_tsvector('public.dutch', name)
	@@ to_tsquery('public.dutch', 'twee');
    name     
-------------
 twee
 tweehonderd
 tweeduizend
(3 rows)

DROP INDEX name_idx;
CREATE INDEX name_idx ON table1 USING GIST (to_tsvector('public.dutch', "name"));
SELECT * FROM table1 WHERE to_tsvector('public.dutch', name)
	@@ to_tsquery('public.dutch', 'deurtje');
  name   
---------
 deuren
 deurtje
 deur
(3 rows)

SELECT * FROM table1 WHERE to_tsvector('public.dutch', name)
	@@ to_tsquery('public.dutch', 'twee');
    name     
-------------
 twee
 tweehonderd
 tweeduizend
(3 rows)

